src/battle/action_cmd/power_shock.c

```diff
@@ -2,12 +2,321 @@
+                    actionCommandStatus->unk_44 +=
+                        battleStatus->unk_434[actionCommandStatus->unk_50] * 850 / 100 * actionCommandStatus->unk_64 / 100;
+                } else {
+                    actionCommandStatus->unk_44 += 100;
+                    if (actionCommandStatus->unk_44 >= 500) {
+                        actionCommandStatus->unk_44 = 500;
+                    }
+                }
+            }
+
+            if (actionCommandStatus->unk_44 > 10000) {
+                hudElement = actionCommandStatus->hudElements[3];
+                actionCommandStatus->unk_44 = 10000;
+                actionCommandStatus->unk_68 = 1;
+                set_hud_element_render_pos(
+                    hudElement,
+                    actionCommandStatus->hudElementX + 50,
+                    actionCommandStatus->hudElementY + 28);
+                clear_hud_element_flags(hudElement, 2);
+            }
+
+            battleStatus->unk_84 = actionCommandStatus->unk_44 / 100;
+            sfx_adjust_env_sound_params(0x80000041, 0, 0, battleStatus->unk_84 * 12);
+
+            if (actionCommandStatus->unk_54 == 0) {
+                phi_a1 = actionCommandStatus->unk_44;
+                if (actionCommandStatus->unk_64 == 0) {
+                    phi_a1 = 0;
+                }
+
+                battleStatus->unk_84 = phi_a1 / 100;
+                if (phi_a1 == 0) {
+                    battleStatus->actionSuccess = -1;
+                } else if (battleStatus->unk_84 >= actionCommandStatus->unk_46) {
+                    battleStatus->actionSuccess = 1;
+                } else {
+                    battleStatus->actionSuccess = 0;
+                }
+
+                if ((actionCommandStatus->mashMeterCutoffs[actionCommandStatus->mashMeterIntervals - 1] / 2) < battleStatus->unk_84) {
+                    battleStatus->unk_86 = 1;
+                } else {
+                    battleStatus->unk_86 = -4;
+                }
+
+                if (battleStatus->actionSuccess == 1) {
+                    func_80269160();
+                }
+
+                sfx_stop_sound(0x80000041);
+                btl_set_popup_duration(0);
+                actionCommandStatus->unk_54 = 5;
+                actionCommandStatus->state = 12;
+            } else {
+                actionCommandStatus->unk_54--;
+            }
+            break;
+        case 12:
+            if (actionCommandStatus->unk_64 == 0) {
+                actionCommandStatus->unk_44 -= 100;
+                if (actionCommandStatus->unk_44 < 0) {
+                    actionCommandStatus->unk_44 = 0;
+                }
+            }
+            if (actionCommandStatus->unk_54 != 0) {
+                actionCommandStatus->unk_54--;
+                break;
+            }
+            func_80268C9C();
+            break;
+    }
+
+    // Invisible "run away"-like meter that determines success threshold.
```

Comment:
Might be funny to link to [this video](https://youtu.be/FagUpV0c67s) here :shipit: 
We can also name unk_46 "meterValue" or something.

Reply:
Added a link in the comment.